brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -110,11 +136,23 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         req = req.withHeaders(newHeaders);
         ctx.updateRequest(req);
 
-        // For no-op spans, we only need to inject into headers and don't set any other attributes.
+        if (scopeDecoratorAdded && !span.isNoop() && ctx instanceof DefaultClientRequestContext) {
+            final DefaultClientRequestContext defaultCtx = (DefaultClientRequestContext) ctx;
+            // Run the scope decorators when the ctx is pushed to the thread local.
+            defaultCtx.hook(() -> currentTraceContext.decorateScope(span.context(),
+                                                                    CLIENT_REQUEST_DECORATING_SCOPE)::close);
```

Comment:
How about using the code block? because it's bit hard for me to understand. 😉 
```suggestion
            defaultCtx.hook(() -> {
                final Scope scope = currentTraceContext.decorateScope(span.context(),
                                                                      CLIENT_REQUEST_DECORATING_SCOPE);
                return scope::close;
            });
```